compiler/modules/types/list.bal

```diff
@@ -572,11 +560,56 @@ function listSubtypeIntersect(Context cx, SubtypeData t1, SubtypeData t2) return
 }
 
 function listSubtypeDiff(Context cx, SubtypeData t1, SubtypeData t2) returns SubtypeData {
-    Bdd b = memoSubtypeDiff(cx.listMemo, <Bdd>t1, <Bdd>t2);
-    if b != false && cx.listMemo[b] == () {
-        //io:println("fresh BDD: ", bddToString(b));
+    MemoBddCache cache = new (cx.listMemo);
+    // This is checking if t1 and t2 are disjoint (i.e intersection(t1, t2) is empty). If so diff(t1, t2) == t1.
+    if listIsEmptySimple(cx, bddIntersect(cache, <Bdd> t1, <Bdd> t2)) == true {
```

Comment:
@jclark we can do this check similar to `simpleIntersect` in the [other PR](https://github.com/ballerina-platform/nballerina/blob/d12899fb5bf882448e0c7f0be895a2e9b6f2b37d/compiler/modules/types/bdd.bal#L85). Not sure which one is more preferable (do that change similar to this i.e in `listSubtypeIntersect` or do this check similar to that in `bddDiff`

Reply:
I think it's better for the complexity to be in the functions that test emptiness.